### PR TITLE
Switch http libraries used in checks to requests

### LINKED ISSUE #1196: Embedded httplib2 does not appear request the correct certificate, triggers DOWN state
It appears that sites hosted on a service that uses a subjectAltName to enable many sites to use SSL on the same IP address, like Fastly or WPEngine, will send along the certificate when the `Host` header matches a site they provide.

When setting `disable_ssl_validation` flag to false, we see this error:

```console
Server presented certificate that does not match host www.datadoghq.com: {'notAfter': 'May 19 22:01:42 2018 GMT', 'subjectAltName': (('DNS', '*.wpengine.com'), ('DNS', 'wpengine.com')), 'subject': ((('organizationalUnitName', u'GT41552380'),), (('organizationalUnitName', u'See www.rapidssl.com/resources/cps (c)14'),), (('organizationalUnitName', u'Domain Control Validated - RapidSSL(R)'),), (('commonName', u'*.wpengine.com'),))}. Connection failed after 128 ms
```

This is due to the remote server responding with a certificate that is not the certificate requested.

Here's a command to retrieve the remote certificate (you can compare commonName and subjectAltName).

    echo | openssl s_client -connect www.datadoghq.com:443 -servername www.datadoghq.com 2>&1 |sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout

The `-servername` flag is the equivalent of setting the HTTP `Host` header flag.

I tried setting the `headers` hash to contain the host header I was requesting, but get the same response.

```python
>>> from httplib2 import Http, HttpLib2Error
>>> h = Http()
>>> resp, content = h.request("https://www.datadoghq.com/", "GET", headers={"host": "www.datadoghq.com", 'user-agent': 'anything'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/httplib2/__init__.py", line 1593, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/httplib2/__init__.py", line 1335, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/httplib2/__init__.py", line 1300, in _conn_request
    conn.connect()
  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/httplib2/__init__.py", line 1032, in connect
    'host %s: %s' % (hostname, cert), hostname, cert)
httplib2.CertificateHostnameMismatch: Server presented certificate that does not match host www.datadoghq.com: {'notAfter': 'May 19 22:01:42 2018 GMT', 'subjectAltName': (('DNS', '*.wpengine.com'), ('DNS', 'wpengine.com')), 'subject': ((('organizationalUnitName', u'GT41552380'),), (('organizationalUnitName', u'See www.rapidssl.com/resources/cps (c)14'),), (('organizationalUnitName', u'Domain Control Validated - RapidSSL(R)'),), (('commonName', u'*.wpengine.com'),))}
```

### LINKED ISSUE #1398: SNI support for http_check missing.
We've serveral SSL/TLS enabled vhosts with different domain names on one machine and IP address using SNI. However, it looks like http_check.py and the http lib it uses do not support SNI.

### PR DESCRIPTION
**Pull request by remh**
Fixes

* Fix #1196
* Fix #1398

Along with the self contained agent (pyopenssl etc) it will support SNI.

See https://github.com/kennethreitz/requests/issues/749 for more
information

### CODE REVIEW

File: checks.d/lighttpd.py [hunk 1]
```diff
@@ -138,17 +141,15 @@ def check(self, instance):
...
-        self.log.debug("Lighttpd server version is Unknown")
-        return "Unknown"
+        if "server" not in headers:
+            self.log.debug("Lighttpd server version is Unknown")
+            return "Unknown"
 
+        v = headers['server']
+        try:
+            version = int(v.split('/')[1][0])
+        except Exception, e:
```
Comment (LeoCavaille) [inline checks.d/lighttpd.py]: or
```python
version = int(headers.get('server').split('/')[1][0])
```
None will raise an exception and you get `Unknown` too?

**Comment by LeoCavaille** [comment]
Thanks for doing this refactoring, this is awesome.
Barely anything to say, it looks good from what I have seen, maybe I'd be even more confident if we had tests for all these checks, but eh we'll see that in the near future.
:fireworks: :heart_eyes_cat: :hammer:

File: checks.d/lighttpd.py (hunk 1 above)
Comment (LeoCavaille) [inline checks.d/lighttpd.py]: or better :factory: match this against a regex? :star2:

Comment (remh) [inline checks.d/lighttpd.py]: Done

**Comment by LeoCavaille** [comment]
BIM!

**Pull request merged**